pkg/glusterutils/exporterd.go

```diff
@@ -13,25 +14,14 @@ var (
 	peerIDPattern = regexp.MustCompile("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}")
 )
 
-// Peers return a list of peers
-func Peers(config *Config) ([]Peer, error) {
-	setDefaultConfig(config)
-
-	if config.GlusterMgmt == MgmtGlusterd2 {
-		return peersGD2(config)
-	}
-
-	return peersGD1(config)
-}
-
 // IsLeader returns true or false based on whether the node is the leader of the cluster or not
-func IsLeader(config *Config) (bool, error) {
-	setDefaultConfig(config)
-	peerList, err := Peers(config)
+func (g GD1) IsLeader(config *Config) (bool, error) {
+	setDefaultConfig(g.config)
+	peerList, err := g.Peers()
 	if err != nil {
 		return false, err
 	}
-	peerID, err := LocalPeerID(config)
+	peerID, err := LocalPeerID(g.config)
```

Comment:
Also add `LocalPeerID` to interface as mentioned earlier. and use `peerID, err := g.LocalPeerID()`